kubernetes/samples/scripts/create-weblogic-domain/domain-home-on-pv/wlst/istio-fix-domain.py

```diff
@@ -0,0 +1,131 @@
+# Licensed under the Universal Permissive License v 1.0 as shown at http://oss.oracle.com/licenses/upl.
+
+def getEnvVar(var):
+  val=os.environ.get(var)
+  if val==None:
+    print "ERROR: Env var ",var, " not set."
+    sys.exit(1)
+  return val
+
+domain_uid                   = getEnvVar("DOMAIN_UID")
+server_port                  = int(getEnvVar("MANAGED_SERVER_PORT"))
+domain_path                  = getEnvVar("DOMAIN_HOME")
+cluster_name                 = getEnvVar("CLUSTER_NAME")
+admin_server_name            = getEnvVar("ADMIN_SERVER_NAME")
+admin_server_name_svc        = getEnvVar("ADMIN_SERVER_NAME_SVC")
+admin_port                   = int(getEnvVar("ADMIN_PORT"))
+domain_name                  = getEnvVar("DOMAIN_NAME")
+t3_channel_port              = int(getEnvVar("T3_CHANNEL_PORT"))
+t3_public_address            = getEnvVar("T3_PUBLIC_ADDRESS")
+number_of_ms                 = int(getEnvVar("CONFIGURED_MANAGED_SERVER_COUNT"))
+cluster_type                 = getEnvVar("CLUSTER_TYPE")
+managed_server_name_base     = getEnvVar("MANAGED_SERVER_NAME_BASE")
+managed_server_name_base_svc = getEnvVar("MANAGED_SERVER_NAME_BASE_SVC")
+domain_logs                  = getEnvVar("DOMAIN_LOGS_DIR")
+script_dir                   = getEnvVar("CREATE_DOMAIN_SCRIPT_DIR")
+
+# Update Domain for Istio
+readDomain(domain_path)
+
+cd('/Servers/' + admin_server_name)
+create('istio-probe', 'NetworkAccessPoint')
+cd('/Servers/%s/NetworkAccessPoints/istio-probe' % admin_server_name)
+set('Protocol', 'http')
+set('ListenAddress', '127.0.0.1')
+set('PublicAddress', domain_uid + '-' + admin_server_name)
+set('ListenPort', 8888)
+set('HttpEnabledForThisProtocol', true)
+set('TunnelingEnabled', false)
+set('OutboundEnabled', false)
+set('Enabled', true)
+set('TwoWaySslEnabled', false)
+set('ClientCertificateEnforced', false)
+
+cd('/Servers/' + admin_server_name)
+create('istio-t3', 'NetworkAccessPoint')
+cd('/Servers/%s/NetworkAccessPoints/istio-t3' % admin_server_name)
+set('ListenAddress', '127.0.0.1')
+set('PublicAddress', domain_uid + '-' + admin_server_name)
+set('ListenPort', admin_port)
+set('TunnelingEnabled', false)
+set('OutboundEnabled', false)
+set('Enabled', true)
+set('TwoWaySslEnabled', false)
+set('ClientCertificateEnforced', false)
+
+cd('/Servers/' + admin_server_name)
+create('istio-ldap', 'NetworkAccessPoint')
+cd('/Servers/%s/NetworkAccessPoints/istio-ldap' % admin_server_name)
+set('Protocol', "ldap")
+set('ListenAddress', '127.0.0.1')
+set('PublicAddress', domain_uid + '-' + admin_server_name)
+set('ListenPort', admin_port)
+set('HttpEnabledForThisProtocol', true)
+set('TunnelingEnabled', false)
+set('OutboundEnabled', false)
+set('Enabled', true)
+set('TwoWaySslEnabled', false)
+set('ClientCertificateEnforced', false)
+
+cd('/Servers/%s/NetworkAccessPoints/T3Channel' % admin_server_name)
+set('ListenAddress', '127.0.0.1')
+set('PublicAddress', domain_uid + '-' + admin_server_name)
```

Comment:
(1) For all addresses, the name needs to be converted to be DNS legal - see introspectDomain.py for example usage:
```  
def toDNS1123Legal(self, address):
    return address.lower().replace('_','-')
``` 
(2) Ideally we should not be setting addresses in the original configuration.  Instead, we should be setting the ListenAddress and Public Address of istio enabled channels for the user via situational configuration generated in 'introspectDomain.py' instead of requiring customers to set them exactly correctly.
(3) An issue is that the public-address of a T3Channel needs to be appropriate for the client's location, so "domain_uid + '-' + admin_server_name" can be wrong if the client is external.   This is also a problem for (2) - as we should not generate an automatic override for PublicAddress if the user is going to put something else there.